lib/restaurant.rb

```diff
@@ -0,0 +1,56 @@
+class Restaurant 
+  attr_reader :order_history, :order, :text
+  attr_accessor :calc
+
+  def initialize(order_class = Order, text_class = Text, sender_class = Sender)
+    @order_class = order_class
+    @order_history = []
+    @order = @order_class.new
+    @calc = 0
+    @text_class = text_class
+    @text = nil
+    @sender_class = sender_class
```

Comment:
`@text_class` and `@sender_class` are only being used within the initialize function, so probably don't need to be stored in an instance variable.